### PR TITLE
Added ability to inject custom API mocks on app launch

### PR DESCRIPTION
**Pull request by treatwell-marius**
In order to use **custom API response mocks** (with special data) we used 
```swift
start(using: AppConfiguration())

app.replaceValues(
    of: [
        "result": "NOT_AUTHENTICATED"
    ],
    in: Stub.Authentication.success
)
```
But API mocked responses were replaced only after next refresh/API call. So as a workaround we had to do additional UI actions like pull to refresh, closing and opening screens again and so on, just to get that API endpoint called again.

This **PR** adds nicer way to do that by introducing `initiationClosure` param to `start(using:)`
So, now there is a way to do:

```swift
start(using: Configuration()) { app in
    app.replaceValues(
        of: [
            "result": "NOT_AUTHENTICATED"
        ],
        in: Stub.Authentication.success
    )
}
.loginStep.authenticationStatusIsVisible()
.loginStep.authenticationStatusIs(equal: "NOT_AUTHENTICATED")
```

Using this closure we get our custom replaced API mocks right after UI Tests launches the app.

To better illustrate that, I have updated example app with few examples.

### CODE REVIEW

File: TWUITests/Components/UITestCase.swift [hunk 1]
```diff
@@ -43,11 +44,25 @@ extension UITestCase: ApplicationStarter {
...
+    }
+
+    /// Start app using specified configuration
+    /// - Parameters:
+    ///   - configuration: Configuration object
+    ///   - initiationClosure: Initiation closure. It is called right after setting up mock server, before app launch.
+    ///   Use this param if you need to inject custom mocked API responses on app launch
+    /// - Returns: UITestApplication
+    @discardableResult
+    public func start(using configuration: Configuration, initiationClosure: ((UITestApplication) -> Void)? = nil) -> UITestApplication {
```
Comment (treatwell-ignasurbonas) [inline TWUITests/Components/UITestCase.swift:64]: Maybe leave posibility to provide `APIStubInfo` or [`APIStubInfo`] as it is in deprecated method?

**Review by treatwell-ignasurbonas — APPROVED** [review]
Looks lovely!

File: TWUITests/Components/UITestCase.swift (hunk 1 above)
Comment (treatwell-marius) [reply TWUITests/Components/UITestCase.swift:64]: Well, it makes no sense to provide `APIStubInfo` or `[APIStubInfo]` in this method as it's completely duplicates what `initiationClosure` is made for. You can do anything you want in this closure, including:
```swift
start(using: Configuration()) { app in
    app.serverUpdate(with: APIStubInfo)
}
```

What do you think @treatwell-ignasurbonas ?

Comment (treatwell-ignasurbonas) [reply TWUITests/Components/UITestCase.swift:64]: Ok, cool let's leave just closure then 👍

**Pull request merged**